widgets/common/src/filters/RuleRow.tsx

```diff
@@ -1,16 +1,16 @@
-import type { FunctionComponent } from 'react';
+import type { ComponentProps, FunctionComponent } from 'react';
 import RuleAttributeDropdown from './RuleAttributeDropdown';
 import RuleOperatorDropdown from './RuleOperatorDropdown';
 import RuleInput from './RuleInput';
 import RuleRemoveButton from './RuleRemoveButton';
-import { FormField, FormGroup } from './FormWrapper';
 
 interface RuleRowProps {
-    id: number;
-    onRemove: (id: number) => void;
+    allowRemove: boolean;
+    onRemove: ComponentProps<typeof Button>['onClick'];
 }
 
-const RuleRow: FunctionComponent<RuleRowProps> = ({ id, onRemove }) => {
+const RuleRow: FunctionComponent<RuleRowProps> = ({ allowRemove = true, onRemove }) => {
```

Comment:
Do we need a default value for a required prop?

```suggestion
const RuleRow: FunctionComponent<RuleRowProps> = ({ allowRemove, onRemove }) => {
```

Reply:
Removed default value setting.